packages/reporter/src/runnables/runnable-and-suite.tsx

```diff
@@ -11,14 +11,20 @@ import Collapsible from '../collapsible/collapsible'
 
 import SuiteModel from './suite-model'
 import TestModel from '../test/test-model'
+import runnablesStoreDefault, { RunnablesStore } from './runnables-store'
 
 interface SuiteProps {
   model: SuiteModel
+  runnablesStore?: RunnablesStore
 }
 
-const Suite = observer(({ model }: SuiteProps) => {
+const Suite = observer(({ model, runnablesStore = runnablesStoreDefault }: SuiteProps) => {
   if (!model.shouldRender) return null
 
+  if (runnablesStore.activeFilter !== null && runnablesStore.activeFilter !== model.state) {
+    return null
+  }
```

Comment:
ℹ️ This is the actual filtering: the `Suite` item will render only if:
- there's no active filter
- its actual state matches the active filter value (from the TestState type)

I wanted to move this inside the Runnable, where we set `shouldRender`, but it seems that we don't have the actual suite state at that place (see https://github.com/cypress-io/cypress/blob/c9719ba733e73f66323de3d7377b9b35613d56df/packages/reporter/src/runnables/runnables-store.ts#L130)